lib/std/fs/path.zig

```diff
@@ -1318,3 +1333,552 @@ test "stem" {
     try testStem(" ", " ");
     try testStem("", "");
 }
+
+/// A path component iterator that can move forwards and backwards.
+/// The 'root' of the path (`/` for POSIX, things like `C:\`, `\\server\share\`, etc
+/// for Windows) is treated specially and will not be iterated over when iterating backwards.
```

Comment:
Please define what "iterating backwards" mean. Does `next` iterate backwards? Also: Is `first()` the root path?